grai-server/app/api/mutations.py

```diff
@@ -73,6 +75,25 @@ async def updateConnection(
 
         return connection
 
+    @strawberry.mutation(permission_classes=[IsAuthenticated])
+    async def runConnection(
+        self,
+        info: Info,
+        connectionId: strawberry.ID,
+    ) -> Connection:
+        user = get_user(info)
+        connection = await sync_to_async(ConnectionModel.objects.get)(pk=connectionId)
```

Comment:
If I were an authenticated user who knew the connectionId of a connection belonging to a separate workspace could I trigger a run from here?

Reply:
Yes you could, there is a general security hole on almost all the mutations for this which needs fixing.

Reply:
Resolved